### ISSUE TITLE
Not enough reserved memory in debug component's ErrorHandler

### ISSUE THREAD

**Issue by krixon** [symfony/symfony#26893]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.4.5

Upon reaching the configured memory_limit, I get a fatal error from the Debug component as it tries to handle the error. For example:

    Fatal error:  Allowed memory size of 67108864 bytes exhausted (tried to allocate 28672 bytes) in /var/www/vendor/symfony/symfony/src/Symfony/Component/Debug/ExceptionHandler.php on line 239

I can fix this by increasing the amount of reserved memory in `Symfony\Component\Debug\ErrorHandler`:

    self::$reservedMemory = str_repeat('x', 20480);

This allows the error handling process to complete and the OutOfMemoryException is thrown as expected, but there is currently no way to configure this value.

**Comment by nicolas-grekas** [comment]
Would you like opening a PR doing so? Doubling the amount of reserved memory looks reasonable to me (did you try any lower value btw?)

**Comment by nicolas-grekas** [comment]
@krixon shall we close then?

**Comment by krixon** [comment]
Sorry, I haven't had time to progress this. I am happy to close, I will reopen if I get to a point where I can reproduce this reliably.

**Comment by jinraynor1** [comment]
Hi, i am using PHP 5.3.3  and loading exception for OutOfMemory exceeds reservedMemory variable, the following snippets prints in my server 46368, look at this example:

```php
$a = memory_get_usage();
include_once 'vendor/symfony/debug/Symfony/Component/Debug/Exception/FatalErrorException.php';
include_once 'vendor/symfony/debug/Symfony/Component/Debug/Exception/OutOfMemoryException.php';
$c = new \Symfony\Component\Debug\Exception\OutOfMemoryException("dummy",1,1,1,1,1);

$b = memory_get_usage();
var_dump($b- $a); // it prints 46368 bytes
exit;
```

> If you attach a logger via method setDefaultLogger then you also need to increase the reservedMemory to your own needs.


I have tried to extend the class and init my own reserved memory with no luck as the ErrorHandler class registers the shutdown function in a no extendable  way
```php
register_shutdown_function(__CLASS__.'::handleFatalError');
```

My ugly solution is copy the all ErrorHandler class and  add my own constant  :(
```php
self::$reservedMemory = str_repeat('x', MEMORY_RESERVED_ERROR_HANDLER);
```

**Comment by jinraynor1** [comment]
I have created a pull request https://github.com/symfony/debug/pull/8

**Comment by sakalys** [comment]
Hi all, this should get fixed in symfony 4.4 soon, as there is a merged in PR for that https://github.com/symfony/symfony/pull/44327
